frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardGraph.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { QueryBrowser } from '@console/internal/components/monitoring/query-browser';
+import { humanizeBinaryBytes } from '@console/internal/components/utils';
+import { ByteDataTypes } from '@console/shared/src/graph-helper/data-utils';
+import { Area } from '@console/internal/components/graphs/area';
+import './MonitoringDashboardGraph.scss';
+
+interface MonitoringDashboardGraphProps {
+  title: string;
+  query: string;
+  namespace: string;
+  graphType?: string;
+}
+
+const defaultTimespan = 30 * 60 * 1000;
+
+const MonitoringDashboardGraph: React.FC<MonitoringDashboardGraphProps> = ({
+  query,
+  namespace,
+  title,
+  graphType = 'area',
+}) => {
+  return (
+    <div className="monitoring-dashboard-graph">
+      {graphType === 'line' ? (
+        <>
+          <h5 className="graph-title">{title}</h5>
+          <QueryBrowser
+            hideControls
+            defaultTimespan={defaultTimespan}
+            namespace={namespace}
+            queries={[query]}
+          />
+        </>
+      ) : (
+        <Area
+          title={title}
+          humanize={humanizeBinaryBytes}
+          byteDataType={ByteDataTypes.BinaryBytes}
```

Comment:
humanize & DataType shall vary & depend on query so it's better to drive both through props.

Reply:
Done